src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -528,90 +541,431 @@ internal override void Complete()
+        }
 
-                        //Wait for outputReader to finish
-                        if (_outputReader != null)
+        /// <summary>
+        /// Read the output from the native process and send it down the line.
+        /// </summary>
+        private void ConsumeAvailableNativeProcessOutput(bool blocking)
+        {
+            if (_isRunningInBackground == false)
+            {
+                if (_nativeProcess.StartInfo.RedirectStandardOutput || _nativeProcess.StartInfo.RedirectStandardError)
+                {
+                    ProcessOutputObject record;
+                    while ((record = DequeueProcessOutput(blocking)) != null)
+                    {
+                        if (this.Command.Context.CurrentPipelineStopping)
                         {
-                            _outputReader.Done();
+                            this.StopProcessing();
+                            return;
                         }
 
-                        // Capture screen output if we are transcribing
-                        if (this.Command.Context.EngineHostInterface.UI.IsTranscribing &&
-                            scrapeHostOutput)
-                        {
-                            Host.Coordinates endPosition = this.Command.Context.EngineHostInterface.UI.RawUI.CursorPosition;
-                            endPosition.X = this.Command.Context.EngineHostInterface.UI.RawUI.BufferSize.Width - 1;
+                        ProcessOutputRecord(record);
+                    }
+                }
+            }
+        }
 
-                            // If the end position is before the start position, then capture the entire buffer.
-                            if (endPosition.Y < startPosition.Y)
-                            {
-                                startPosition.Y = 0;
-                            }
+        internal override void Complete()
+        {
+            Exception exceptionToRethrow = null;
+            try
+            {
+                if (_isRunningInBackground == false)   
+                {
+                    //Wait for input writer to finish.
+                    _inputWriter.Done();
 
-                            Host.BufferCell[,] bufferContents = this.Command.Context.EngineHostInterface.UI.RawUI.GetBufferContents(
-                                new Host.Rectangle(startPosition, endPosition));
+                    //Wait for the process to exit and consume available output
+                    while (!_nativeProcess.HasExited)
+                    {
+                        ConsumeAvailableNativeProcessOutput(blocking: true);
+                    }
 
-                            StringBuilder lineContents = new StringBuilder();
-                            StringBuilder bufferText = new StringBuilder();
+                    // read all the available output one more time
+                    _nativeProcess.WaitForExit();
+                    ConsumeAvailableNativeProcessOutput(blocking: true);
```

Comment:
This `ConsumeAvailableNativeProcessOutput(blocking: true);` won't be necessary. One blocking reading of the queue is guaranteed to read all output/error out of it.